### PR TITLE
Roles

### PR DESCRIPTION
**Pull request by tux255**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,8 +1,22 @@
...
 
-  has_many :articles
+  private
+
+  def set_default_role
+    self.role ||= :user
+  end
+
+  def create
+    @user = User.new(params[:user].permit(:id, :email, :password, :role))
```
Comment (igor04) [inline app/models/user.rb]: I think it doesn't work, because model has no `params`, this is responsibility of controller to `permit` params, while model could have validation

File: app/policies/article_policy.rb [hunk 2]
```diff
@@ -1,4 +1,23 @@
 class ArticlePolicy < ApplicationPolicy
+  class Scope
```
Comment (igor04) [inline app/policies/article_policy.rb]: I think it's more convenient to place Scope at the bottom of policy

File: app/services/create_admin_service.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class CreateAdminService
+  def call
+    user = User.find_or_create_by!(email: Rails.application.secrets.admin_email) do |user|
```
Comment (igor04) [inline app/services/create_admin_service.rb:3]: why do we need to create admin users? I think user could be registered and then simply promoted by SuperAdmin to Admin or another role

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
+  # after_action :verify_authorized
+  before_action :configure_permitted_parameters, if: :devise_controller?
 
   rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
 
-  # after_action :verify_authorized
+  protected
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: %i[name email password role])
```
Comment (igor04) [inline app/controllers/application_controller.rb]: we don't need to receive `role` as a parameter from a client, because it always could be `SuperDuperAdmin`

File: app/controllers/articles_controller.rb [hunk 5]
```diff
@@ -3,7 +3,8 @@ class ArticlesController < ApplicationController
 
   # GET /articles
   def index
-    @articles = Article.all
+    @articles = Article.all.where(published: true)
```
Comment (igor04) [inline app/controllers/articles_controller.rb]: looks like `.where(published: true)` expects to be handled by policy

File: app/controllers/articles_controller.rb [hunk 6]
```diff
@@ -3,7 +3,8 @@ class ArticlesController < ApplicationController
...
+    @current_user = current_user
```
Comment (igor04) [inline app/controllers/articles_controller.rb]: I think `current_user` is already available in views without this assignment

File: app/models/user.rb (hunk 1 above)
Comment (tux255) [reply app/models/user.rb]: removed create method

File: app/policies/article_policy.rb (hunk 2 above)
Comment (tux255) [reply app/policies/article_policy.rb]: fixed

File: app/services/create_admin_service.rb (hunk 3 above)
Comment (tux255) [reply app/services/create_admin_service.rb:3]: Now I have three types of users:
- admin -> superadmin
- user - will have less abilities than admin
- guest - do not exist. the person who visited website and can only read contents

File: app/controllers/application_controller.rb (hunk 4 above)
Comment (tux255) [reply app/controllers/application_controller.rb]: fixed

File: app/services/create_admin_service.rb (hunk 3 above)
Comment (tux255) [reply app/services/create_admin_service.rb:3]: This service is used with db:seed to create superadmin

Comment (igor04) [reply app/services/create_admin_service.rb:3]: you don't need code in `app` folder which is used only from `db/seed`

Comment (igor04) [reply app/services/create_admin_service.rb:3]: instead of `Rails.application.secrets.admin_password` you could just have `password = "test"` in your seed file

File: app/views/devise/registrations/edit.html.erb [hunk 7]
```diff
@@ -1,8 +1,13 @@
 <h2>Edit <%= resource_name.to_s.humanize %></h2>
-
+<%= current_user.admin? %>
```
Comment (igor04) [inline app/views/devise/registrations/edit.html.erb:2]: use could edit own information, so not sure why this condition is needed

File: app/views/layouts/_navigation.html.erb [hunk 8]
```diff
@@ -1,21 +1,21 @@
 <nav class="navbar navbar-inverse p-0 position-absolute justify-content-between w-100">
-  <div class="container-fluid p-0">
-    <div class="navbar-header px-5 py-3">
-      <%= link_to 'Sports Blog', root_path, class: 'navbar-brand m-0' %>
+  <div class="container">
+    <div class="navbar-header">
+      <%= link_to 'Pundit Blog', root_path, class: 'navbar-brand' %>
```
Comment (igor04) [inline app/views/layouts/_navigation.html.erb:4]: looks like unnecessary change from `Sports` to `Pundit`

File: app/views/layouts/application.html.erb [hunk 9]
```diff
@@ -8,14 +8,14 @@
   </head>
 
   <body>
-    <%= render "layouts/navigation" %>
+    <%= render user_signed_in? ? 'layouts/navigation' : 'layouts/sign_navigation' %>
+    <div class="container-fluid">
+      <%= yield %>
+    </div>
     <div id="flash">
       <% flash.each do |key, value| %>
```
Comment (igor04) [inline app/views/layouts/application.html.erb:16]: flash messages are expected to be under navigation not at the bottom

File: db/seeds.rb [hunk 10]
```diff
@@ -1,7 +1,19 @@
 # This file should contain all the record creation needed to seed the database with its default values.
 # The data can then be loaded with the bin/rails db:seed command (or created alongside the database with db:setup).
 #
-# Examples:
-#
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+require 'faker'
+
+user = CreateAdminService.new.call
```
Comment (igor04) [inline db/seeds.rb:6]: check comment about this service https://github.com/tux255/sports_hub_portal/pull/2/files#r641578308

File: app/views/layouts/_navigation.html.erb (hunk 8 above)
Comment (tux255) [reply app/views/layouts/_navigation.html.erb:4]: it is typo

File: app/services/create_admin_service.rb (hunk 3 above)
Comment (tux255) [reply app/services/create_admin_service.rb:3]: replaced with recommended solution

File: db/seeds.rb (hunk 10 above)
Comment (tux255) [reply db/seeds.rb:6]: checked and solved
removed secrets.yml and service completely.

File: app/views/devise/registrations/edit.html.erb (hunk 7 above)
Comment (tux255) [reply app/views/devise/registrations/edit.html.erb:2]: removed junk check

File: app/views/layouts/_navigation.html.erb (hunk 8 above)
Comment (tux255) [reply app/views/layouts/_navigation.html.erb:4]: solved

File: app/views/layouts/application.html.erb (hunk 9 above)
Comment (tux255) [reply app/views/layouts/application.html.erb:16]: solved

File: db/seeds.rb (hunk 10 above)
Comment (tux255) [reply db/seeds.rb:6]: solved. Removed service.
